os/pm/pm_timedsuspend.c

```diff
@@ -79,17 +79,33 @@ static WDOG_ID pid_timer_map[CONFIG_MAX_TASKS];
  * Private Functions
  ************************************************************************/
 
-static void timer_timeout(int argc, int domain_id, uint32_t hash_pid)
+static void timer_timeout(int argc, int domain_id, uint32_t pid)
 {
-	DEBUGASSERT(pid_timer_map[hash_pid] != NULL);
+	DEBUGASSERT(pid_timer_map[pid] != NULL);
 	/* PM transition will be resume here */
 	if (pm_resume(domain_id) != OK) {
 		pmlldbg("Unable to resume domain: %s\n", pm_domain_map[domain_id]);
 	}
-	(void)wd_delete(pid_timer_map[hash_pid]);
-	pid_timer_map[hash_pid] = NULL;
+	(void)wd_delete(pid_timer_map[pid]);
+	pid_timer_map[pid] = NULL;
 }
 
+static void pm_timedsuspend_cancel(void)
+{
+	WDOG_ID wdog;
+	uint32_t pid;
+	irqstate_t flags;
+	/* Don't let actual timer to expire during this process */
+	flags = enter_critical_section();
+	pid = getpid();
+	wdog = pid_timer_map[pid];
+	if (wdog) {
+		timer_timeout(2, wdog->parm[0], pid);
```

Comment:
What is the hardcoded value 2? Can we either use a preprocessor definition for it or add a comment?

Reply:
`2` is no. of argument

Reply:
Ok